irony-cdb.el

```diff
@@ -63,6 +63,20 @@ for files that it cannot handle."
   :type '(repeat function)
   :group 'irony-cdb)
 
+(defcustom irony-cdb-search-directory-list '("." "build")
```

Comment:
I just saw this, guess we made the right choice:

$ cat cmake/utils.cmake 
 
> function(check_for_in_source_build)
>   if (CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR AND NOT MSVC_IDE)
>     message(FATAL_ERROR "In-source builds are not allowed.
>
>   **Please create a build/ directory** and run cmake from there, passing
>   the path to this source directory as the last argument. This process
>   created the file `CMakeCache.txt' and the directory `CMakeFiles'.
>   Please delete them.
> ")
>   endif()
> endfunction()

Looks like it really is the "official" place to build things.